Civi/Api4/Generic/Traits/SavedSearchInspectorTrait.php

```diff
@@ -282,14 +282,24 @@ protected function applyFilter($fieldName, $value) {
           $filterClauses[] = ['AND', $andGroup];
         }
       }
-      elseif (!empty($field['serialize']) && in_array('CONTAINS', $operators, TRUE)) {
-        $filterClauses[] = [$fieldName, 'CONTAINS', $value];
+      elseif (!empty($field['serialize'])) {
+        if (in_array('CONTAINS', $operators, TRUE)) {
+          $filterClauses[] = [$fieldName, 'CONTAINS', $value];
+        }
+        if (in_array('NOT CONTAINS', $operators, TRUE)) {
+          $filterClauses[] = [$fieldName, 'NOT CONTAINS', $value];
```

Comment:
The way to reach this code is to create a search that includes a serialized field (e.g. Group.parents) and then make a search Afform which includes that field as an exposed filter.

What this code does is apply the default filter when no explicit operator is given, and the default is "CONTAINS". I don't think the changes you've made in this file are necessary, as this is just setting defaults which should remain unchanged by this PR. The change you've made here actually looks like it would break the filter because it would add two contradictory clauses to the query (a field cannot both CONTAIN and NOT CONTAIN a value!)

Reply:
OK, thanks. This was the cause of test failures, so I think you're right.